services/httpd/handler_test.go

```diff
@@ -284,6 +284,65 @@ func TestHandler_Query_ErrResult(t *testing.T) {
 	}
 }
 
+// Ensure the handler handles ping requests correctly.
+func TestHandler_Ping(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	h.ServeHTTP(w, MustNewRequest("GET", "/ping", nil))
+	if w.Code != http.StatusNoContent {
+		t.Fatalf("unexpected status: %d", w.Code)
+	}
+	h.ServeHTTP(w, MustNewRequest("HEAD", "/ping", nil))
+	if w.Code != http.StatusNoContent {
+		t.Fatalf("unexpected status: %d", w.Code)
+	}
+}
+
+// Ensure the handler handles ping requests correctly, when waiting for leader.
+func TestHandler_PingWaitForLeader(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	h.ServeHTTP(w, MustNewRequest("GET", "/ping?wait_for_leader=1s", nil))
```

Comment:
Well, at the risk of bike-shedding, I think we should differentiate between the path i.e. ping and a query param. I also prefer hyphens in the paths, but query params seem different to me. For example, Twitter uses underscores in its query params.